pyanaconda/modules/common/structures/subscription.py

```diff
@@ -362,3 +362,107 @@ def server_proxy_password(self) -> SecretData:
     @server_proxy_password.setter
     def server_proxy_password(self, password: SecretData):
         self._server_proxy_password = password
+
+
+class AttachedSubscription(DBusData):
+    """Data for a single attached subscription."""
+
+    def __init__(self):
+        self._name = ""
+        self._service_level = ""
+        self._sku = ""
+        self._contract = ""
+        self._start_date = ""
+        self._end_date = ""
+        self._consumed_entitlement_count = ""
+
+    @property
+    def name(self) -> Str:
+        """Name of the attached subscription.
+
+        :return: subscription name
+        :rtype: str
+        """
+        return self._name
+
+    @name.setter
+    def name(self, name: Str):
+        self._name = name
+
+    @property
+    def service_level(self) -> Str:
+        """Service level of the attached subscription.
+
+        :return: service level
+        :rtype: str
+        """
+        return self._service_level
+
+    @service_level.setter
+    def service_level(self, service_level: Str):
+        self._service_level = service_level
+
+    @property
+    def sku(self) -> Str:
+        """SKU id of the attached subscription.
+
+        :return: SKU id
+        :rtype: str
+        """
+        return self._sku
+
+    @sku.setter
+    def sku(self, sku: Str):
+        self._sku = sku
+
+    @property
+    def contract(self) -> Str:
+        """Contract identifier.
+
+        :return: contract identifier
+        :rtype: str
+        """
+        return self._contract
+
+    @contract.setter
+    def contract(self, contract: Str):
+        self._contract = contract
+
+    @property
+    def start_date(self) -> Str:
+        """Subscription start date.
+
+        :return: start date of the subscription
+        :rtype: str
+        """
+        return self._start_date
+
+    @start_date.setter
+    def start_date(self, start_date: Str):
+        self._start_date = start_date
+
+    @property
+    def end_date(self) -> Str:
+        """Subscription end date..
+
+        :return: end date of the subscription
+        :rtype: str
+        """
+        return self._end_date
+
+    @end_date.setter
+    def end_date(self, end_date: Str):
+        self._end_date = end_date
+
+    @property
+    def consumed_entitlement_count(self) -> Str:
```

Comment:
Shouldn't this be an integer?

Reply:
The idea is that we are not really sure - the JSON returned by the RHSM DBus API from which this data is returned has what seems to be a string containing an integer. But given that the format is not well documented and I fear we might one day get "3 entitlements" or something like that.

So it seems prudent to me to treat this data as an opaque string that we do not really interpret to avoid surprises later on.

Reply:
This is one of the bugs for the JSON output not being documented: [bug 1790987](https://bugzilla.redhat.com/show_bug.cgi?id=1790987)

Reply:
In that case, could you document the explanation in the docstring of the property?

Reply:
So I looked deeper into this and in the end it looks rather unlikely we will get something else than an integer in the key we are parsing in the JSON file. Also the "consumed" suffix is added by us at the moment. So I've switched the property to be an `Int`.